s3tests_boto3/functional/test_s3.py

```diff
@@ -13166,3 +13169,108 @@ def test_multipart_upload_on_a_bucket_with_policy():
     (upload_id, data, parts) = _multipart_upload(bucket_name=bucket_name, key=key, size=objlen, client=client)
     response = client.complete_multipart_upload(Bucket=bucket_name, Key=key, UploadId=upload_id, MultipartUpload={'Parts': parts})
     eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+@attr(resource='bucket')
+@attr(method='get')
+@attr(operation='get cors configuration')
+@attr(assertion='with no cors set on  a bucket, fails 404')
+def test_get_cors_no_cors():
+    bucket = get_new_bucket()
+    e = assert_raises(ClientError, bucket_get_cors, bucket_name)
+    status, error_code = _get_status_and_error_code(e.response)
+    eq(status, 404)
+
+@attr(resource='bucket')
+@attr(method='get')
+@attr(operation='get cors configuration')
+@attr(assertion='with non-existant bucket, fails')
+def test_get_cors_no_bucket():
+    bucket_name = "non_existant_bucket"
+    e = assert_raises(ClientError, bucket_get_cors, bucket_name)
+    status, error_code = _get_status_and_error_code(e.response)
+    eq(status, 404)
+
+PUT_ORIGIN = "www.*.com"
+OTHER_ORIGIN = "*.com"
+
+ALLOWED_HEADERS = ["*"]
+ALLOWED_METHODS = ['PUT', 'POST', 'GET']
+ALLOWED_ORIGINS = [PUT_ORIGIN]
+EXPOSE_HEADERS = ['*']
+MAX_AGE_SECONDS = 123
+
+cors_config = {
+    'CORSRules': [
+        {
+            'AllowedHeaders': ALLOWED_HEADERS,
+            'AllowedMethods': ALLOWED_METHODS,
+            'AllowedOrigins': ALLOWED_ORIGINS,
+            'ExposeHeaders': EXPOSE_HEADERS,
+            'MaxAgeSeconds': MAX_AGE_SECONDS
+        },
+    ]
+}
+
+@attr(resource='bucket')
+@attr(method='get')
+@attr(operation='get cors configuration')
+@attr(assertion='get/set cors of a bucket, returns 200')
+def test_get_cors_correct_rules():
```

Comment:
split this test?
1. first test should only validate set
2. 2nd test does set+get